### PR TITLE
NPA-3332 Improve Repository Documentation

### PR DESCRIPTION
**Pull request by JackPlowman**
## Ticket Link

https://nhsd-jira.digital.nhs.uk/browse/NPA-3332

## Description/Change Summary

- `README.md` has been rewritten for a external consumer perspective
- `DEVELOPMENT_GUIDE.md` has been created to host the development/testing/releasing documentation for the schema, proxies and sandbox
- `CONTRIBUTING.md` has been reformatted (with prettier)

### CODE REVIEW

File: DEVELOPMENT_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+      - [`manifest_template.yml`:](#manifest_templateyml)
+      - [Package management:](#package-management)
+  - [Releasing a new schema version](#releasing-a-new-schema-version)
+
+## Development
+
+### Requirements
+
+-   GNU make
+-   nodejs + npm/yarn
```
Comment (miiisterjim) [inline DEVELOPMENT_GUIDE.md:43]: Worth indicating the minimum Node version

File: DEVELOPMENT_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+$ make install
+```
+
+#### Updating hooks
+
+You can install some pre-commit hooks to ensure you can't commit invalid spec changes by accident. These are also run
+in CI, but it's useful to run them locally too.
+
+```
+$ make install
```
Comment (miiisterjim) [inline DEVELOPMENT_GUIDE.md:59]: Is this intentionally the same command as the install section above?

File: DEVELOPMENT_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+There are `make` commands that alias some of this functionality:
+
+-   `lint` -- Lints the spec and code
+-   `publish` -- Outputs the specification as a **single file** into the `build/` directory
+-   `serve` -- Serves a preview of the specification in human-readable format
+
+### Testing
+
+Each API and team is unique. We encourage you to use a `test/` folder in the root of the project, and use whatever testing frameworks or apps your team feels comfortable with. It is important that the URL your test points to be configurable. We have included some stubs in the Makefile for running tests.
```
Comment (miiisterjim) [inline DEVELOPMENT_GUIDE.md:78]: This section feels like boilerplate from the APIM template. Is there concrete guidance on testing for our setup we can give?

File: DEVELOPMENT_GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+As currently defined in your `proxies` folder, your proxies do pretty much nothing.
+Telling Apigee how to connect to your backend requires a _Target Server_, which you should call named `validated-relationships-service-api-target`.
+Our _Target Servers_ defined in the [api-management-infrastructure](https://github.com/NHSDigital/api-management-infrastructure) repository.
+
+:bulb: For Sandbox-running environments (`test`) these need to be present for successful deployment but can be set to empty/dummy values.
+
+### Detailed folder walk through
+
+To get started developing your API use this template repo alongside guidance provided by the [API Producer Zone confluence](https://nhsd-confluence.digital.nhs.uk/display/APM/Deliver+your+API)
```
Comment (miiisterjim) [inline DEVELOPMENT_GUIDE.md:123]: Reword or remove given it pertains to the template and external contributors won't have access to the API producer docs in confluence (may well be fine to link to it, but perhaps a note to call out that it's internal NHSE guidance documentation)

File: DEVELOPMENT_GUIDE.md (hunk 2 above)
Comment (JackPlowman) [reply DEVELOPMENT_GUIDE.md]: Removed the dedicated section as that install is used for testing. Added minor sections to each area that has it's own install

File: DEVELOPMENT_GUIDE.md [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+
+### Caveats
+
+#### Swagger UI
+
+Swagger UI unfortunately doesn't correctly render `$ref`s in examples, so use `speccy serve` instead.
+
+#### Apigee Portal
+
+The Apigee portal will not automatically pull examples from schemas, you must specify them manually.
```
[the remark this answers is not in the document]
Comment (JackPlowman) [reply DEVELOPMENT_GUIDE.md]: Searched confluence for documentation and was unable to find it as such I've removed this section

File: DEVELOPMENT_GUIDE.md (hunk 4 above)
Comment (JackPlowman) [reply DEVELOPMENT_GUIDE.md]: At the top of the document I've added a warning to state that the development guide contains links to NHS internal documentation

**Pull request merged**